File: JavaSource/com/yuanluesoft/jeaf/sso/forms/ChangePassword.java
package com.yuanluesoft.jeaf.sso.forms;

import com.yuanluesoft.jeaf.form.ActionForm;

/**
 * 
 * @author linchuan
 *
 */
public class ChangePassword extends ActionForm {
	private String userName; //用户名
	private int passwordStrength; //密码强度
	private String oldPassword; //旧密码
	private String newPassword; //新密码
	
	/**
	 * @return the newPassword
	 */
	public String getNewPassword() {
		return newPassword;
	}
	/**
	 * @param newPassword the newPassword to set
	 */
	public void setNewPassword(String newPassword) {
		this.newPassword = newPassword;
	}
	/**
	 * @return the oldPassword
	 */
	public String getOldPassword() {
		return oldPassword;
	}
	/**
	 * @param oldPassword the oldPassword to set
	 */
	public void setOldPassword(String oldPassword) {
		this.oldPassword = oldPassword;
	}
	/**
	 * @return the userName
	 */
	public String getUserName() {
		return userName;
	}
	/**
	 * @param userName the userName to set
	 */
	public void setUserName(String userName) {
		this.userName = userName;
	}
	/**
	 * @return the passwordStrength
	 */
	public int getPasswordStrength() {
		return passwordStrength;
	}
	/**
	 * @param passwordStrength the passwordStrength to set
	 */
	public void setPasswordStrength(int passwordStrength) {
		this.passwordStrength = passwordStrength;
	}
}

File: WebContent/jeaf/sso/js/changePassword.js
function formOnSubmit() {
	if(document.getElementsByName("oldPassword")[0].value=='') {
		alert("请输入原密码！");
		document.getElementsByName("oldPassword")[0].focus();
		return false;
	}
	if(document.getElementsByName("newPassword")[0].value=='') {
		alert("请输入新密码！");
		document.getElementsByName("newPassword")[0].focus();
		return false;
	}
	if(document.getElementsByName("newPassword")[0].value!=document.getElementsByName("newPassword")[1].value) {
		alert("密码不一致！");
		document.getElementsByName("newPassword")[1].focus();
		return false;
	}
	var passwordStrength = Number(document.getElementsByName("passwordStrength")[0].value);
	if(passwordStrength>1 && getPasswordStrength(document.getElementsByName("newPassword")[0].value)<passwordStrength) {
		alert("密码安全等级不够！");
		document.getElementsByName("newPassword")[0].focus();
		return false;
	}
	return true;
}